server/service/client.go

```diff
@@ -233,6 +233,20 @@ func (c *Client) CheckMDMEnabled() error {
 	return nil
 }
 
+func (c *Client) CheckPremiumMDMEnabled() error {
+	appCfg, err := c.GetAppConfig()
+	if err != nil {
+		return err
+	}
+	if !appCfg.MDM.EnabledAndConfigured {
+		return errors.New("MDM features aren't turned on. Use `fleetctl generate mdm-apple` and then `fleet serve` with `mdm` configuration to turn on MDM features.")
+	}
+	if appCfg.License == nil || appCfg.License.Tier != fleet.TierPremium {
+		return errors.New("missing or invalid license")
+	}
```

Comment:
probably a nit: should we check the license first? reasoning is that if the feature is not meant to be used with the current license, I wouldn't want to point the user to try to configure MDM and either go back here and then get the license error OR get the license error while provisioning the ABM certs.